Source/Core/writeTextToCanvas.js

```diff
@@ -1,9 +1,95 @@
-import measureText from "../ThirdParty/measureText.js";
 import Color from "./Color.js";
 import defaultValue from "./defaultValue.js";
 import defined from "./defined.js";
 import DeveloperError from "./DeveloperError.js";
 
+function measureText(context2D, textString, font, stroke, fill) {
+  var metrics = context2D.measureText(textString);
+  var isSpace = !/\S/.test(textString);
+
+  if (!isSpace) {
+    var fontSize = document.defaultView
+      .getComputedStyle(context2D.canvas)
+      .getPropertyValue("font-size")
+      .replace("px", "");
+    var canvas = document.createElement("canvas");
+    var padding = 100;
+    var width = (metrics.width + padding) | 0;
+    var height = 3 * fontSize;
+    var baseline = height / 2;
+    canvas.width = width;
+    canvas.height = height;
+
+    var ctx = canvas.getContext("2d");
+    ctx.font = font;
+    ctx.fillStyle = "white";
+    ctx.fillRect(0, 0, canvas.width + 1, canvas.height + 1);
+
+    if (stroke) {
+      ctx.strokeStyle = "black";
+      ctx.lineWidth = context2D.lineWidth;
+      ctx.strokeText(textString, padding / 2, baseline);
+    }
+
+    if (fill) {
+      ctx.fillStyle = "black";
+      ctx.fillText(textString, padding / 2, baseline);
+    }
+
+    // Context image data has width * height * 4 elements, because
+    // each pixel's R, G, B and A are consecutive values in the array.
+    var pixelData = ctx.getImageData(0, 0, width, height).data;
+    var length = pixelData.length;
+    var width4 = width * 4;
+    var i, j;
+
+    var ascent, descent;
+    // Find the number of rows (from the top) until the first non-white pixel
+    for (i = 0; i < length; ++i) {
+      if (pixelData[i] !== 255) {
+        ascent = (i / width4) | 0;
+        break;
+      }
+    }
+
+    // Find the number of rows (from the bottom) until the first non-white pixel
+    for (i = length - 1; i > 0; --i) {
+      if (pixelData[i] !== 255) {
+        descent = (i / width4) | 0;
+        break;
+      }
+    }
+
+    var minx = -1;
+    // For each column, for each row, check for first non-white pixel
+    for (i = 0; i < width && minx === -1; ++i) {
+      for (j = 0; j < height; ++j) {
+        if (
+          pixelData[i * 4 + j * width4] !== 255 ||
+          pixelData[i * 4 + j * width4 + 1] !== 255 ||
+          pixelData[i * 4 + j * width4 + 2] !== 255 ||
+          pixelData[i * 4 + j * width4 + 3] !== 255
+        ) {
+          minx = i;
+          break;
+        }
```

Comment:
Factor out the index into its own variable since it's used multiple times

```suggestion
        var pixelIndex = i * 4 + j * width4;
        if (
          pixelData[pixelIndex] !== 255 ||
          pixelData[pixelIndex + 1] !== 255 ||
          pixelData[pixelIndex + 2] !== 255 ||
          pixelData[pixelIndex + 3] !== 255
        ) {
          minx = i;
          break;
        }
```